chaco/legend.py

```diff
@@ -355,7 +358,7 @@ def get_preferred_size(self):
         if len(self.plots) == 0:
             return [0, 0]
 
-        plot_names, visible_plots = map(list, zip(*sorted(self.plots.items())))
+        plot_names, visible_plots = list(sm.map(list, sm.zip(*sorted(six.iteritems(self.plots)))))
```

Comment:
Your approach will not work. You need to have the items() in there

```python
x = {'b': 1, 'a': 3}

assert sorted(x) == ['a', 'b']
assert sorted(x.items()) == [('a', 3), ('b', 1)]
```

For standard python contructs. See my comment in enthought/traits#374

It's more or less the same issue. I would prefer to keep the iterator version whenever possible.

Let me know your decision.